apiml/src/main/java/org/zowe/apiml/handler/SuccessTicketHandler.java

```diff
@@ -0,0 +1,143 @@
+            return writeResponse(webFilterExchange, response, messageView, HttpStatus.BAD_REQUEST);
+        });
+    }
+
+    private Mono<Void> writeResponse(WebFilterExchange webFilterExchange, ServerHttpResponse response, ApiMessageView messageView, HttpStatusCode statusCode) {
+        var bufferFactory = response.bufferFactory();
+        DataBuffer buffer;
+        try {
+            buffer = bufferFactory.wrap(mapper.writeValueAsBytes(messageView));
+        } catch (JsonProcessingException e) {
+            throw new IncorrectRequestBodyException("ApplicationName not provided");
+        }
+        response.setStatusCode(statusCode);
+        log.debug("The generation of the PassTicket failed. Please supply a valid user and application name, and check that corresponding permissions have been set up.");
+        return webFilterExchange.getExchange().getResponse().writeWith(Mono.just(buffer));
+    }
+
+    private Mono<TicketResponse> getTicketResponse(ServerHttpRequest request, Authentication authentication) {
+        TokenAuthentication tokenAuthentication = (TokenAuthentication) authentication;
+        var userId = tokenAuthentication.getPrincipal();
+        return request.getBody().collectList().handle((bufferList, sink) -> {
+            var bufferFactory = new DefaultDataBufferFactory();
+            int totalSize = bufferList.stream().mapToInt(DataBuffer::readableByteCount).sum();
+            DataBuffer mergedBuffer = bufferFactory.allocateBuffer(totalSize);
+            bufferList.forEach(mergedBuffer::write); // Write each buffer into the new one
+
+            String applicationName;
+            try {
+                applicationName = mapper.readValue(mergedBuffer.asInputStream(), TicketRequest.class).getApplicationName();
+                if (applicationName == null || applicationName.trim().isEmpty()) {
+                    sink.error(new IncorrectRequestBodyException("ApplicationName not provided"));
+                    return;
+                }
+            } catch (IOException e) {
+                sink.error(new IncorrectRequestBodyException("ApplicationName not provided"));
+                return;
+            }
+
+            String ticket;
+            try {
+                ticket = passTicketService.generate(userId, applicationName);
+            } catch (PassTicketException e) {
+                sink.error(new PassException(e));
+                return;
+            }
+
+            sink.next(new TicketResponse(tokenAuthentication.getCredentials(), userId, applicationName, ticket));
+        });
+    }
+}
+
+class IncorrectRequestBodyException extends RuntimeException {
+    public IncorrectRequestBodyException(String message) {
+        super(message);
+    }
+}
+
+class PassException extends RuntimeException {
+    final PassTicketException e;
+
+    public PassException(PassTicketException e) {
+        this.e = e;
```

Comment:
This is not a cause?

Reply:
do you mean passing directly the cause rather than the exception? I am not sure I understand

Reply:
I would assume to call `super(e);` to set the cause. The exception also does not have a field `serialVersionUID` as an implementation of `Serializable` interface.

Actually, there is just one usage:
```java
var ex = (IRRPassTicketGenerationException) exception.e;
```

It could be replaced with 
```java
var ex = (IRRPassTicketGenerationException) exception.getCause();
```

And the field could be removed at all.